src/test/ui/inference/deref-suggestion.stderr

```diff
@@ -87,7 +87,10 @@ error[E0308]: mismatched types
   --> $DIR/deref-suggestion.rs:37:5
    |
 LL |     assert_eq!(3i32, &3i32);
-   |     ^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `&i32`
+   |     ^^^^^^^^^^^^^^^^^^^^^^^
+   |     |
+   |     expected `i32`, found `&i32`
+   |     expected because of this binary operation left hand side expression
```

Comment:
I find this to be confusing. Can we not add this note if the binop is within a macro expansion, or maybe find a way to actually point at `3i32` here?

Reply:
We can't point at the macro argument because the macro machinery doesn't have enough intelligence yet to do that once its gone though any expression. In this case the macro expands to something that loses the span. That is a project that I've tried to tackle in the past and will not be fixed anytime soon, sadly.